DataFormats/NanoAOD/interface/MergeableCounterTable.h

```diff
@@ -0,0 +1,103 @@
+#ifndef DataFormats_NanoAOD_MergeableCounterTable_h
+#define DataFormats_NanoAOD_MergeableCounterTable_h
+
+#include <FWCore/Utilities/interface/Exception.h>
+#include <vector>
+#include <string>
+
+class MergeableCounterTable {
+    public:
+        MergeableCounterTable() {}
+        typedef long long int_accumulator; // we accumulate in long long int, to avoid overflow
+        typedef double float_accumulator; // we accumulate in double, to preserve precision
+
+        template<typename T>
+        struct SingleColumn {
+            typedef T value_type;
+            SingleColumn() {}
+            SingleColumn(const std::string & aname, const std::string & adoc, T avalue = T()) : name(aname), doc(adoc), value(avalue) {}
+            std::string name, doc;
+            T value;
+            void operator+=(const SingleColumn<T> & other) {
+                //if (name != other.name) throw cms::Exception("LogicError", "Trying to merge "+name+" with "+other.name+"\n");
+                value += other.value;
+            }
+            bool compatible(const SingleColumn<T> & other) {
+                return name == other.name;  // we don't check the doc, not needed
+            }
+        };
+        typedef SingleColumn<float_accumulator> FloatColumn; 
+        typedef SingleColumn<int_accumulator> IntColumn;  
+       
+        template<typename T>
+        struct VectorColumn {
+            typedef T element_type;
+            VectorColumn() {}
+            VectorColumn(const std::string & aname, const std::string & adoc, unsigned int size) : name(aname), doc(adoc), values(size, T()) {}
+            VectorColumn(const std::string & aname, const std::string & adoc, const std::vector<T> & somevalues) : name(aname), doc(adoc), values(somevalues) {}
+            std::string name, doc;
+            std::vector<T> values;
+            void operator+=(const VectorColumn<T> & other) {
+                //if (name != other.name) throw cms::Exception("LogicError", "Trying to merge "+name+" with "+other.name+"\n");
+                //if (values.size() != other.values.size()) throw cms::Exception("LogicError", "Trying to merge "+name+" with different number of values!\n");
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant